### PR TITLE
EVG-6170 turn on new task queue service as an experiment

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: service/api_task_test.go [hunk 1]
```diff
@@ -163,108 +163,227 @@ func TestAssignNextAvailableTask(t *testing.T) {
...
+		//
+		// model.TaskQueueService
+		//
+		// type TaskQueueService interface {
+		// 	FindNextTask(string, TaskSpec) (*TaskQueueItem, error)
+		// 	Refresh(string) error
+		// 	RefreshFindNextTask(string, TaskSpec) (*TaskQueueItem, error)
+		// }
+
+		////////////////////////////////////////////////////////////////////////////
```
Comment (tychoish) [inline service/api_task_test.go]: uncomment this?

**Comment by bsamek** [comment]
@StuartButler Is this ready for review, or are there specific parts that you're interested in feedback on?

**Comment by StuartButler** [comment]
@bsamek - it's ready for review.

File: model/task_queue_service.go [hunk 2]
```diff
@@ -146,20 +157,7 @@ func shouldRefreshCached(ttl time.Duration, lastUpdated time.Time) bool {
...
-	if shouldRefreshCached(t.ttl, t.lastUpdated) {
-		t.lastUpdated = time.Now()
-		return true
-	}
-
-	return false
-}
-
 func (t *taskDistroDispatchService) rebuild(items []TaskQueueItem) {
-	if !shouldRefreshCached(t.ttl, t.lastUpdated) {
```
Comment (bsamek) [inline model/task_queue_service.go]: Is there still a ttl that prevents this from being rebuilt very frequently?

File: service/api_task_test.go (hunk 1 above)
Comment (StuartButler) [reply service/api_task_test.go]: New tests added.

**Pull request merged**